### DISCUSSION TITLE
get records againt id

### DISCUSSION THREAD

**Discussion by msamran27** [filamentphp/filament#6590] [General]
i want to show records aganist id i get data against id in $timeSlot variable but not showeid please help me
the code is 
    public static function table(Table $table): Table
    {
        $id = request()->query('id');

        if ($id) {
            $daySlot = DaySlot::findOrFail($id); // Retrieve the specific DaySlot record
            $timeSlots = TimeSlot::where('day_id', $daySlot->id)->get();
            return $table
                ->columns([
                    Tables\Columns\TextColumn::make('id')->searchable(),
                    Tables\Columns\TextColumn::make('dayslot.day', 'Day Name')->searchable(),
                    Tables\Columns\TextColumn::make('from')->searchable(),
                    Tables\Columns\TextColumn::make('to')->searchable(),
                ])
                ->filters([])
                ->actions([
                    Tables\Actions\EditAction::make(),
                    Tables\Actions\DeleteAction::make(),
                ])
                ->bulkActions([
                    Tables\Actions\DeleteBulkAction::make(),
                ]);
        }

        return $table
            ->columns([
                Tables\Columns\TextColumn::make('id')->searchable(),
                Tables\Columns\TextColumn::make('dayslot.day', 'Day Name')->searchable(),
                Tables\Columns\TextColumn::make('from')->searchable(),
                Tables\Columns\TextColumn::make('to')->searchable(),
            ])
            ->filters([])
            ->actions([
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }

**Comment by msamran27** [comment]
Please Response

**Reply by msamran27** [reply]
?

**Comment by danharrin** [comment]
I don't understand what you want to do. Please be patient when waiting for an answer from someone who does understand.

**Reply by msamran27** [reply]
i want to get data aganist request id ,data show in datble
$id = request->query('id) 
    if ($id) {
        $daySlot = DaySlot::findOrFail($id); // Retrieve the specific DaySlot record
        $timeSlots = TimeSlot::where('day_id', $daySlot->id)->get();
        return $table
            ->columns([
                Tables\Columns\TextColumn::make('id')->searchable(),
                Tables\Columns\TextColumn::make('dayslot.day', 'Day Name')->searchable(),
                Tables\Columns\TextColumn::make('from')->searchable(),
                Tables\Columns\TextColumn::make('to')->searchable(),
            ])
            ->filters([])
            ->actions([
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\DeleteBulkAction::make(),
            ]);
    }

**Reply by msamran27** [reply]
how to apply i query in this column

**Reply by msamran27** [reply]
![image](https://github.com/filamentphp/filament/assets/100018533/2d21fd8d-a393-4aed-a8bd-caed99f897c8)
how to apply where condition in it 
i want show data aganist req id

**Reply by danharrin** [reply]
You don't need to do anything inside `table()`. You probably want to override `getTableQuery()` on the List page of the resource, and add a `where()` with the `request('id')` to the query.

**Reply by msamran27** [reply]
Thanks you it's working
